sw/device/lib/uart.c

```diff
@@ -35,6 +35,10 @@ int uart_tx_empty(void) {
   return !!(REG32(UART_STATUS(0)) & (1 << UART_STATUS_TXEMPTY));
 }
 
+int uart_tx_idle(void) {
```

Comment:
If this is getting turned into a 1 or 0 anyway can this just be `bool`?

Reply:
I didn't nit this because "rule of local style" trumps "rule of weird". If doing this change, I think all the functions in this TU which are supposed to be `bool` but are declared as `int` should be changed... and also, `!!x` should be changed to the much clearer `x != 0`.

Reply:
yeah to be honest i thought `x != 0` was clearer also, but like @mcy, just followed local style.
I'll follow up with another PR later.